File: packages/usf-ts/src/adapters/ollama.ts
import type { CompiledPrompt } from "../prompt.js";

const DEFAULT_MODEL = "llama3.1:8b";

export function render(compiled: CompiledPrompt, opts: { model?: string } = {}): Record<string, unknown> {
  const fm = compiled.frontmatter;
  const hints = (fm["model_hints"] as Record<string, string> | undefined) ?? {};
  const model = opts.model ?? hints.ollama ?? DEFAULT_MODEL;
  const options: Record<string, unknown> = {};
  if ("recommended_temperature" in fm) options.temperature = fm["recommended_temperature"];
  if ("max_tokens" in fm) options.num_predict = fm["max_tokens"];
  const payload: Record<string, unknown> = {
    model,
    system: compiled.system,
    prompt: compiled.user,
    stream: false,
  };
  if (Object.keys(options).length) payload.options = options;
  return payload;
}

File: packages/usf-py/usf/adapters/ollama.py
"""Ollama local-LLM adapter (/api/generate style)."""
from __future__ import annotations

from ..prompt import CompiledPrompt

DEFAULT_MODEL = "llama3.1:8b"


def render(compiled: CompiledPrompt, *, model: str | None = None) -> dict:
    fm = compiled.frontmatter
    chosen_model = model or (fm.get("model_hints") or {}).get("ollama") or DEFAULT_MODEL
    options: dict = {}
    if "recommended_temperature" in fm:
        options["temperature"] = fm["recommended_temperature"]
    if "max_tokens" in fm:
        options["num_predict"] = fm["max_tokens"]
    payload: dict = {
        "model": chosen_model,
        "system": compiled.system,
        "prompt": compiled.user,
        "stream": False,
    }
    if options:
        payload["options"] = options
    return payload
